torch/distributed/distributed_c10d.py

```diff
@@ -753,6 +784,61 @@ def recv(tensor,
         return src
 
 
+@contextlib.contextmanager
+def _group_manager(backend):
+    if backend == Backend.NCCL:
+        ProcessGroupNCCL.group_start()
+    try:
+        yield
+    finally:
+        if backend == Backend.NCCL:
+            ProcessGroupNCCL.group_end()
+
+
+def batch_isend_irecv(op_list,
+                      tensor_list,
+                      peer_list,
+                      group_list=None,
+                      tag_list=None):
+    """
+    Send or Receive tensors asynchronously and return a list of requests.
+
+    Process the first item in each passed parameters, and then the second item
+    in each passed parameters, etc.
+
+    Arguments:
+        op_list: list of point-to-point operations(type of each operations is either
+        ``torch.distributed.isend`` or ``torch.distributed.irecv``.
+        tensor_list (list[Tensor]): list of send or recv tensors.
+        peer_list (list[int]): list of peer ranks to send to or receive from.
+        group_list (list[ProcessGroup], Optional): list of groups to operator on.
+        tag_list (list[int], Optional): list of tags to match send with recv.
+
+    Return:
+        a list of distributed request objects returned by calling the corresponding
+        op in the op_list.
+    """
+    _check_op_list(op_list, "op_list")
+    _check_input_length(op_list, tensor_list, peer_list, group_list, tag_list)
+
+    backend = get_backend(group.WORLD if group_list is None else group_list[0])
```

Comment:
If we are planning to support a `group_list`, we should ensure that every group in the list has the same backend and also add this requirement to our docs.